### PR TITLE
Fee Adjustment Tests

### PR DESCRIPTION
**Pull request by gabriele-0201**
Tests are copied from polkadot-sdk/substrate/bin/node/runtime/impls.rs and adapted to test our struct BlobsFeeAdjustment

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/153?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #149** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/149?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #150** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/150?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #151** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/151?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #152** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/152?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #153** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/153?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/153?utm_source=stack-comment).

File: sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+};
+use pallet_transaction_payment::{Multiplier, TargetedFeeAdjustment};
+use sp_runtime::{
+    assert_eq_error_rate,
+    traits::{Convert, One, Zero},
+    BuildStorage, FixedPointNumber,
+};
+
+use crate::{
+    constants::{consensus::*, kusama::currency::*},
```
Comment (pepyakin) [inline sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs:22]: avoid non-prelude glob imports.

File: sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+        t.execute_with(|| {
+            match self {
+                MultiplierType::Length => {
+                    // AllExtrinsicsLen is a private storage value of the system pallet
+                    // so the key must be manually constructed
+                    let len = w.ref_time();
+                    sp_io::storage::set(
+                        &[
+                            sp_core::twox_128(b"System"),
+                            sp_core::twox_128(b"AllExtrinsicsLen"),
```
Comment (pepyakin) [inline sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs:184]: This probably should be upstreamed.

Comment (gabriele-0201) [reply sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs:184]: What do you mean? Are you saying that AllExtrinsicsLen should not be private or that there should be a public helper to set the value of that StorageValue?

Comment (rphmeier) [reply sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs:169]: I believe @pepyakin means an upstream PR to Polkadot-SDK

**Review by rphmeier — APPROVED** [review]

File: sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs (hunk 2 above)
Comment (gabriele-0201) [reply sugondat-chain/runtimes/sugondat-kusama/src/fee_adjustment_tests.rs:169]: Looking into `polkadot-sdk/master/src/frame_system/lib.rs` to understand how many things would be required to make the PR, I've noticed this:

```rust
	pub fn set_block_consumed_resources(weight: Weight, len: usize) {
		BlockWeight::<T>::mutate(|current_weight| {
			current_weight.set(weight, DispatchClass::Normal)
		});
		AllExtrinsicsLen::<T>::put(len as u32); // <-----
	}
```

This means that I don't need to manually construct the key, and I can simply use this method!

**Comment by rphmeier** [comment]
### Merge activity

* **Dec 28, 3:09 PM**: @rphmeier started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/153).

**Pull request merged**